models/DataObject/Service.php

```diff
@@ -1669,6 +1672,21 @@ public static function getCsvDataForObject(Concrete $object, string $requestedLa
         if ($returnMappedFieldNames) {
             $tmp = [];
             foreach ($mappedFieldnames as $key => $value) {
+                $titleTranslation = Translation::getByKey(
+                    $key,
+                    Translation::DOMAIN_ADMIN
+                );
+
+                if (
+                    !empty($titleTranslation)
+                    && $titleTranslation->hasTranslation($requestedLanguage)
+                    && !empty($titleTranslation->getTranslation($requestedLanguage))
+                ) {
+                    $tmp[] = '"' . $titleTranslation->getTranslation($requestedLanguage) . '"';
+
+                    continue;
+                }
+
                 $tmp[$value] = $objectData[$key];
             }
             $objectData = $tmp;
```

Comment:
Hi,

This code get translation for field code, not label. In my opinion it should get translation for $value variable. Moreover tmp key in line 1685 is integer, not translated value so this code rather not work. The proper solution should look like this
```
 if ($returnMappedFieldNames) {
            $tmp = [];

            foreach ($mappedFieldnames as $key => $value) {
                $translation = Translation::getByKey($value, Translation::DOMAIN_ADMIN);
                $translatedValue = $translation && $translation->hasTranslation($requestedLanguage)
                    ? $ translation->getTranslation($requestedLanguage)
                    : null;

                $tmp[$translatedValue ?: $value] = $objectData[$key];
            }

            $objectData = $tmp;
}
```

Reply:
@kevinloeber-twocream could you have a look and come back to me? If you do not have time it is fine, I can also take over if you want to.

Reply:
@mattamon,  unfortunately I don't have that much time right now, could you do it for me?

Reply:
@kevinloeber-twocream, @nbogucki 
So I checked again and tried to apply your solution @nbogucki.
Now the label gets translated accordingly. 
But there is one thing I still find odd and it has nothing to do with your PR in general.

There is a difference between the shown label in the grid and the exported label.
I have two translations in my admin translation table.
![image](https://github.com/user-attachments/assets/99d7ef7d-5585-460f-b6a9-6c38987a7bb8)

Now for the grid it shows Path2
![image](https://github.com/user-attachments/assets/2151726e-c5d5-4d2a-a5c9-0aeffe493ca1)

And for the export it takes the system key and translate this so it is Path1
![image](https://github.com/user-attachments/assets/ad82f385-9dee-4f07-92e9-c5e003e6a965)
Another problem is that the admin translation key is a simple text, so as you can see html tags get exported too. 

So currently I am not too happy with how this works in general. 

WDYT? What should we do with this PR?